daemon/start.go

```diff
@@ -130,7 +130,7 @@ func (daemon *Daemon) containerStart(ctx context.Context, daemonCfg *configStore
 			if err := container.CheckpointTo(daemon.containersReplica); err != nil {
 				log.G(ctx).Errorf("%s: failed saving state on start failure: %v", container.ID, err)
 			}
-			container.Reset(false)
+			container.Reset(context.Background(), false)
```

Comment:
why not pass the context here?

Reply:
I did not pass it only because I tried to be somewhat careful when doing changes in this PR, trying to change how the daemon works only where necessary (at least until explicitly discussed with maintainers)

Reply:
I have updated the code: now `ctx` is passed here. Also, context is now passed to `CloseStreams` in `daemon/exec.go` [here](https://github.com/asterite3/moby/blob/13f9fd2/daemon/exec.go#L190)